arkouda/join.py

```diff
@@ -205,6 +210,29 @@ def inner_join(
     """
     from inspect import signature
 
+    if type(left) != type(right):
+        raise TypeError("Left and Right arrays must be of same type")
+
+    # Convert Strings and Categorical left & right values to Categorical.codes. This generates an
+    # index representation of the strings. This way they can be handled as numerics and follow the
+    # same process as int
+    t = cast(str, pdarray.objtype)
+    cats = None
+
+    if isinstance(left, Strings) and isinstance(right, Strings):
+        left = Categorical(left)
+        right = Categorical(right)
+
+    if isinstance(left, Categorical) and isinstance(right, Categorical):
+        t = cast(str, Categorical.objtype)
+        left, right = Categorical.standardize_categories([left, right])
+        if isinstance(left, Categorical) and isinstance(right, Categorical):
+            cats = left.categories[left.codes]  # Get standardized categories without NAvalue
+            leftKeepCodes = left.isna() == 0  # Invert bool array
+            rightKeepCodes = right.isna() == 0  # Invert bool array
+            left = left.codes[leftKeepCodes]
+            right = right.codes[rightKeepCodes]
```

Comment:
I'm curious how this compares to something like:
```python
left, right = Categorical.standardize_categories([left, right])
# maybe isinstance check for mypy
left.reset_categorizes()
right.reset_categorizes()
```

Reply:
Some tests using `10**6` problem size showed this to be slower than what's currently there.
Current method:
```
-------------------------------------- benchmark 'Inner Join': 1 tests --------------------------------------
Name (time in s)         Min      Max     Mean  StdDev   Median     IQR  Outliers     OPS  Rounds  Iterations
-------------------------------------------------------------------------------------------------------------
bench_inner_join     42.1997  44.3982  42.7200  0.9449  42.2967  0.7485       1;1  0.0234       5           1
-------------------------------------------------------------------------------------------------------------
```

Suggested Method:
```
-------------------------------------- benchmark 'Inner Join': 1 tests --------------------------------------
Name (time in s)         Min      Max     Mean  StdDev   Median     IQR  Outliers     OPS  Rounds  Iterations
-------------------------------------------------------------------------------------------------------------
bench_inner_join     47.6216  49.4303  48.4438  0.7423  48.4373  1.2332       2;0  0.0206       5           1
-------------------------------------------------------------------------------------------------------------
```